tests/integration/targets/aci_l3out/tasks/main.yml

```diff
@@ -0,0 +1,255 @@
+
+- name: Execute tasks only for non-cloud sites
+  when: query_cloud.current == []  # This condition will execute only non-cloud sites
+  block:  # block specifies execution of tasks within, based on conditions
+  - name: Add a new tenant
+    aci_tenant:
+      <<: *aci_info 
+      tenant: ansible_tenant
+      description: Ansible tenant
+      state: present
+
+  - name: Remove the ansible_l3out
+    aci_l3out: &aci_l3out_absent
+      <<: *aci_info
+      tenant: ansible_tenant
+      name: ansible_l3out
+      description: L3Out for ansible_tenant tenant
+      domain: ansible_dom
+      vrf: ansible_vrf
+      state: absent
+
+  - name: Remove the second ansible_l3out
+    aci_l3out: &aci_l3out_2_absent
+      <<: *aci_info
+      tenant: ansible_tenant
+      name: ansible_l3out_2
+      description: Second L3Out
+      domain: ansible_dom
+      vrf: ansible_vrf
+      l3protocol: ospf
+      state: absent
+  
+  - name: Add a new L3Out (check_mode)
+    aci_l3out: &aci_l3out_present
+      <<: *aci_l3out_absent
+      state: present
+    check_mode: true
+    register: cm_add_l3out
+
+  - name: Add a new L3Out (normal_mode)
+    aci_l3out:
+      <<: *aci_l3out_present
+    register: nm_add_l3out
+  
+  - name: create L3Out again - idempotency works
+    aci_l3out:
+      <<: *aci_l3out_present
+    register: create_idempotency
+
+  - name: Add the second ansible_l3out
+    aci_l3out:
+      <<: *aci_l3out_2_absent
+      state: present
+    register: cm_add_l3out_2
+
+  - name: asserts for l3out creation tasks
+    assert:
+      that:
+        - cm_add_l3out is changed
+        - nm_add_l3out is changed
+        - nm_add_l3out.sent.l3extOut.attributes.descr == cm_add_l3out.sent.l3extOut.attributes.descr == "L3Out for ansible_tenant tenant"
+        - nm_add_l3out.sent.l3extOut.attributes.name == cm_add_l3out.sent.l3extOut.attributes.name == "ansible_l3out"
+        - create_idempotency is not changed
+
+  - name: Add export and import to route_control 
+    aci_l3out:
+      <<: *aci_l3out_present
+      route_control: [ import, export ]
+    register: nm_add_l3out_import_export
+
+  - name: Add export to route_control 
+    aci_l3out:
+      <<: *aci_l3out_present
+      route_control: export
+    register: nm_add_l3out_export
+
+  - name: Add just import to route_control - failure message works
+    aci_l3out:
+      <<: *aci_l3out_present
+      route_control: import
+    register: nm_add_l3out_import_error
+    ignore_errors: true
+
+  - name: asserts for l3out route control tasks
+    assert:
+      that:
+        - nm_add_l3out_import_export is changed
+        - nm_add_l3out_import_export.sent.l3extOut.attributes.enforceRtctrl == "import,export"
+        - nm_add_l3out_export is changed
+        - nm_add_l3out_export.sent.l3extOut.attributes.enforceRtctrl == "export"
+        - nm_add_l3out_import_error.msg == "The route_control parameter is invalid: allowed options are export or import,export only"
```

Comment:
The route_control parameter is invalid:  -> The route_control parameter is invalid{{':'}}

else : might cause issue.